site/src/pages/docs/advanced-spring-boot-integration.mdx

```diff
@@ -0,0 +1,382 @@
+# Using Spring Boot with Armeria
+
+Spring framework provides powerful features which are necessary for building a web application, such as
+dependency injection, data binding, AOP, transaction. Armeria is a lightweight microservices framework,
+its capabilities is comparable to existing full stack web frameworks. It has Rich support for
+[Apache Thrift](https://thrift.apache.org/) and [gRPC](https://grpc.io/), including [the fancy web console](https://armeria.dev/docs/server-docservice)
+that enables you to send Thrift and gRPC requests from a web browser. Armeria supports both HTTP/1 and HTTP/2,
+and these protocols both support [cleartext](https://simple.wikipedia.org/wiki/Cleartext) and TLS cryptographic protocols.
+Armeria uses [event loop](https://en.wikipedia.org/wiki/Event_loop) based on [Netty](https://netty.io/index.html)
+which allows to perform non-blocking I/O operations. Therefore, it shows better performance in certain situations such as media streaming,
+or large file transfer. If you integrate your Spring application with Armeria, you can serve both legacy Spring service
+and any service made of Armeria such as gRPC or Thrift in the same port. Also, you can gradually migrate your services
+starting with those expected to improve performance.
+
+Armeria provides a way to integrate legacy Spring Boot applications with just simple code.
+Additionally, Armeria supports several internal services for monitoring and management purposes.
+You can achieve these convenient properties through bean configuration.
+
+## Integrate Legacy Spring Boot with Armeria
```

Comment:
What do you think of dividing this section into two?
One for integrating with Spring Boot and another one for Spring MVC?
If so, we can distinguish the usage of each artifact and explain it more clearly. e.g.
```java
## Integrating with Spring Boot

<Tip>

Visit [armeria-examples](https://github.com/line/armeria/tree/main/examples/spring-boot-minimal) to
find a fully working example.

</Tip>

First, you need the `armeria-spring-boot3-starter` dependency:
...
// Explain how an Armeria server is configured and started using Spring Boot.
// We can also introduce ArmeriaServerConfigurator.

## Integrating with Spring MVC
// tip and dependency section needs.
// Explain how to integrate Spring MVC with Armeria using TomcatService.

```

Reply:
I split the section into two parts. However, I placed the "Integrating with Spring MVC" section first, and then the "Integrating with Spring Boot" section. The readers of this document are likely to be divided into three groups: those using Spring with Tomcat, those using Spring WebFlux, and those using Spring as dependency injection tool. I think most people are using Spring with Tomcat

Therefore, I placed the "Integrating with Spring MVC" section at the beginning. Next, I placed the "Integrating with Spring Boot" section for the group using only Spring dependency injection or Boot's auto configuration. The group using Spring WebFlux is likely to refer to other existing documents.